### PR TITLE
RDK-28664 : Persistent Storage Thunder API

### PR DESCRIPTION
**Pull request by npoltorapavlo**
Reason for change: new thunder plugin
Test Procedure: test functionality specified in docs
Risks: Low
Signed-off-by: Nikita Poltorapavlo <npoltorapavlo@productengine.com>

### CODE REVIEW

File: PersistentStore/PersistentStore.cpp [hunk 1]
```diff
@@ -0,0 +1,709 @@
...
+            {
+                LOGERR("%d : %s", rc, sqlite3_errmsg(db));
+                term();
+                return false;
+            }
+
+            if (shouldEncrypt)
+            {
+#if defined(SQLITE_HAS_CODEC)
+                std::vector<uint8_t> pKey;
```
Comment (mhughesacn) [inline PersistentStore/PersistentStore.cpp:611]: Hi Nikita,
The code is being matched by blackduck and must be used with credit.   Please add a comment before this line as follows:
/* Based on pxCore, Copyright 2015-2018 John Robinson */
/* Licensed under the Apache License, Version 2.0 */

and append the same two lines to the end of NOTICE (in the top level).

Thanks,
Martin
Martin Hughes, RDK CMF Compliance Team

**Comment by npoltorapavlo** [comment]
@mhughesacn your suggestion is not working

**Comment by mhughesacn** [comment]
Hi @npoltorapavlo.  Thank you for making the change.  I still have to approve the match in the tools.  Just to be consistent, the NOTICE file entry doesn't need a C comment style, just format it the same as the other entries.  Sorry, I didn't say that clearly.  Thanks, Martin

**Comment by mhughesacn** [comment]
Thanks @npoltorapavlo for the rapid response. Nothing outstanding from a compliance point of view. - Martin

**Pull request merged**